deltachat-ios/Controller/Settings/Proxy/ShareProxyViewController.swift

```diff
@@ -0,0 +1,142 @@
+import UIKit
+import DcCore
+import SDWebImageSVGKitPlugin
+
+class ShareProxyViewController: UIViewController {
+    private let dcContext: DcContext
+
+    private let contentStackView: UIStackView
+    private let contentScrollView: UIScrollView
+    private let qrContentView: UIImageView
+    private let explanationLabel: UILabel
+    private let shareLinkButton: UIButton
+
+    private let proxyUrlString: String
+
+    var verticalCenterConstraint: NSLayoutConstraint?
+    var contentTopAnchor: NSLayoutConstraint?
+    var contentBottomAnchor: NSLayoutConstraint?
+
+    init(dcContext: DcContext, proxyUrlString: String) {
+        self.dcContext = dcContext
+        self.proxyUrlString = proxyUrlString
+
+        qrContentView = UIImageView()
+        qrContentView.contentMode = .scaleAspectFit
+        qrContentView.translatesAutoresizingMaskIntoConstraints = false
+
+        shareLinkButton = UIButton(type: .system)
+        shareLinkButton.setTitle(String.localized("proxy_share_link"), for: .normal)
+        shareLinkButton.translatesAutoresizingMaskIntoConstraints = false
+
+        explanationLabel = UILabel()
+        explanationLabel.translatesAutoresizingMaskIntoConstraints = false
+        explanationLabel.font = UIFont.preferredFont(forTextStyle: .caption1)
+        explanationLabel.text = String.localized("proxy_share_explain")
+
+        contentStackView = UIStackView(arrangedSubviews: [qrContentView, explanationLabel, shareLinkButton, UIView()])
+        contentStackView.translatesAutoresizingMaskIntoConstraints = false
+        contentStackView.axis = .vertical
+        contentStackView.alignment = .center
+        contentStackView.spacing = 16
+
+        contentScrollView = UIScrollView()
+        contentScrollView.translatesAutoresizingMaskIntoConstraints = false
+        contentScrollView.addSubview(contentStackView)
+
+        super.init(nibName: nil, bundle: nil)
+
+        if #available(iOS 13, *) {
+            view.backgroundColor = .secondarySystemBackground
+        } else {
+            view.backgroundColor = DcColors.defaultBackgroundColor
+        }
+
+        shareLinkButton.addTarget(self, action: #selector(ShareProxyViewController.shareInviteLink(_:)), for: .touchUpInside)
+        shareLinkButton.setTitleColor(DcColors.primary, for: .normal)
+
+        let svg = dcContext.createQRSVG(for: proxyUrlString)
+        qrContentView.image = getQrImage(svg: svg)
+
+        navigationItem.leftBarButtonItem = UIBarButtonItem(title: String.localized("done"), style: .done, target: self, action: #selector(ShareProxyViewController.done(_:)))
```

Comment:
a single button needs to go to the right:

```suggestion
        navigationItem.rightBarButtonItem = UIBarButtonItem(title: String.localized("done"), style: .done, target: self, action: #selector(ShareProxyViewController.done(_:)))
```